Write error detection for "Npm failed with return code: 3221225477"

https://dev.azure.com/azure-sdk/590cfd2a-581c-4dcb-a12e-6568ce786175/_apis/build/builds/639416/logs/37

```
2020-12-02T16:44:04.2863008Z ##[section]Starting: Install Azurite
2020-12-02T16:44:04.2998502Z ==============================================================================
2020-12-02T16:44:04.2999451Z Task         : npm
2020-12-02T16:44:04.3000375Z Description  : Install and publish npm packages, or run an npm command. Supports npmjs.com and authenticated registries like Azure Artifacts.
2020-12-02T16:44:04.3001279Z Version      : 1.178.0
2020-12-02T16:44:04.3001942Z Author       : Microsoft Corporation
2020-12-02T16:44:04.3002776Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/package/npm
2020-12-02T16:44:04.3003677Z ==============================================================================
2020-12-02T16:44:04.7808817Z SYSTEMVSSCONNECTION exists true
2020-12-02T16:44:04.9869431Z SYSTEMVSSCONNECTION exists true
2020-12-02T16:44:05.1644251Z [command]C:\windows\system32\cmd.exe /D /S /C ""C:\Program Files\nodejs\npm.cmd" --version"
2020-12-02T16:44:21.9136713Z 6.14.8
2020-12-02T16:44:23.2999195Z [command]C:\windows\system32\cmd.exe /D /S /C ""C:\Program Files\nodejs\npm.cmd" config list"
2020-12-02T16:44:23.8782401Z ; cli configs
2020-12-02T16:44:23.8784095Z metrics-registry = "https://registry.npmjs.org/"
2020-12-02T16:44:23.8785272Z scope = ""
2020-12-02T16:44:23.8786468Z user-agent = "npm/6.14.8 node/v14.15.0 win32 x64"
2020-12-02T16:44:23.8787081Z 
2020-12-02T16:44:23.8787742Z ; environment configs
2020-12-02T16:44:23.8788473Z cache = "C:\\npm\\cache"
2020-12-02T16:44:23.8793631Z prefix = "C:\\npm\\prefix"
2020-12-02T16:44:23.8794699Z userconfig = "D:\\a\\1\\npm\\639416.npmrc"
2020-12-02T16:44:23.8795292Z 
2020-12-02T16:44:23.8796014Z ; builtin config undefined
2020-12-02T16:44:23.8796517Z 
2020-12-02T16:44:23.8797274Z ; node bin location = C:\Program Files\nodejs\node.exe
2020-12-02T16:44:23.8798092Z ; cwd = D:\a\1\s
2020-12-02T16:44:23.8798847Z ; HOME = C:\Users\VssAdministrator
2020-12-02T16:44:23.8799679Z ; "npm config ls -l" to show all defaults.
2020-12-02T16:44:23.8800221Z 
2020-12-02T16:44:23.8804597Z [command]C:\windows\system32\cmd.exe /D /S /C ""C:\Program Files\nodejs\npm.cmd" install --prefix D:\a\_temp/azurite azurite"
2020-12-02T16:44:37.3674709Z ##[warning]Couldn't find a debug log in the cache or working directory
2020-12-02T16:44:37.3691379Z ##[error]Error: Npm failed with return code: 3221225477
2020-12-02T16:44:37.3697057Z ##[section]Finishing: Install Azurite
```
For reference the failing step is https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/tests-install-azurite.yml. 

@mitchdenny can you add a rule to detect this issue to see how often it occurs.

From a quick look online it isn't clear what causes this problem but it seems like it could be some file access issue, maybe a locked file because of antivirus? As it seems to be mostly a windows issue but I'd like pipeline witness to gather more data to help figure out an what the issue might be.

--- a/tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/AzuriteInstallFailureClassifier.cs
+++ b/tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/AzuriteInstallFailureClassifier.cs
@@ -1,0 +1,30 @@
+﻿using Azure.Sdk.Tools.PipelineWitness.Entities.AzurePipelines;
+using Microsoft.TeamFoundation.Build.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services.FailureAnalysis
+{
+    public class AzuriteInstallFailureClassifier : IFailureClassifier
+    {
+        public async Task ClassifyAsync(FailureAnalyzerContext context)
+        {
+            var failedTasks = from r in context.Timeline.Records
+                                where r.Result == TaskResult.Failed
+                                where r.RecordType == "Task"
+                                where r.Name == "Install Azurite"
+                                select r;
+
+            if (failedTasks.Count() > 0)
+            {
+                foreach (var failedTask in failedTasks)
+                {
+                    context.AddFailure(failedTask, "Azurite Install");
+                }
+            }
+        }
+    }
+}

--- a/tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Startup.cs
+++ b/tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Startup.cs
@@ -73,6 +73,7 @@ namespace Azure.Sdk.Tools.PipelineWitness
             builder.Services.AddMemoryCache();
             builder.Services.AddSingleton<RunProcessor>();
             builder.Services.AddSingleton<IFailureAnalyzer, FailureAnalyzer>();
+            builder.Services.AddSingleton<IFailureClassifier, AzuriteInstallFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, CancelledTaskClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, CosmosDbEmulatorStartFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, AzurePipelinesPoolOutageClassifier>();
@@ -85,7 +86,7 @@ namespace Azure.Sdk.Tools.PipelineWitness
             builder.Services.AddSingleton<IFailureClassifier, JsDevFeedPublishingFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, DownloadSecretsFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, GitCheckoutFailureClassifier>();
-            builder.Services.AddSingleton<IFailureClassifier, AzurePowerShellModuleInstallationFailureClassifier>();
+            builder.Services.AddSingleton<IFailureClassifier, AzuriteInstallFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, MavenBrokenPipeFailureClassifier>();
             builder.Services.AddSingleton<IFailureClassifier, CodeSigningFailureClassifier>();
         }

